int32 type
int32 ID
float32 value
